framework/lib/components/combo-picker/combo-picker.tsx

```diff
@@ -87,10 +90,24 @@ export const ComboPicker = ({
   }, [ enableSelectInput ])
 
   useEffect(() => {
-    if (!is(Number, inputValue) && defaultToZeroIfEmpty) {
-      handleInputChange(clamp(min, max, 0))
+    const needsToCorrectOption = !selectOption
+    const needsToCorrectInput = !is(Number, inputValue) && defaultToZeroIfEmpty
+
+    const option = needsToCorrectOption ? options[0] : selectOption
```

Comment:
Does selectedOption now also have a defaultValue? And what about defaultValue prop instead of having options[0] always as the default

Reply:
Unless I'm misunderstanding your question, we always had a default selectedOption which is options[0], for the case in which the user does not pass one, but now the place where we check if it had a selectedOption when rendered or not is here, and then we assign the default value, it only changed the place where it happens